### PR TITLE
#169258640: Create a route for getting user requests

### PR DESCRIPTION
**Pull request by hezronkimutai**
### What does this PR do?
- This PR gets all the users trip requests.
#### Description of Task to be completed?
- Create a route for getting all the requests of the user by id
#### How should this be manually tested?
`git clone https://github.com/andela/team-odd-bn-backend`
`git checkout ft-get-requests-169258640`
Open postman and test the `\trips` endpoint
#### What are the relevant pivotal tracker stories?
[#169258640](https://www.pivotaltracker.com/n/projects/2407428/stories/169258640)
#### Screenshots (if appropriate)
Screen Shot 2019-11-26 at 10 32 31

#### Questions:
- Who should be able to view all the routes?

### CODE REVIEW

File: src/middlewares/Exists.js [hunk 1]
```diff
@@ -25,6 +26,26 @@ class Exists {
...
+    */
+  static async isUsersRequestExist(req, res, next) {
+    const allRequests = await tripRequests.findAll(
+      { where: { userId: req.user.id }, raw: true }
+    );
+
+    if (allRequests.length > 0) {
+      return next();
+    }
+    return Customize.errorMessage(req, res, 'No request availlable', 404);
```
Comment (hezronkimutai) [inline src/middlewares/Exists.js]: Change this status code.

File: src/middlewares/isUserVerified.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
 import { users } from '../database/models';
 
 const isUserVerified = async (req, res, next) => {
-  const { email } = req.user;
+  const { email } = req.user || req.body;
```
Comment (hezronkimutai) [inline src/middlewares/isUserVerified.js]: Remove req.body

File: src/middlewares/Exists.js [hunk 3]
```diff
@@ -25,6 +26,26 @@ class Exists {
...
+  /**
+    * check is the request exist
+    * @static
+    * @param {object} req request object
+    * @param {object} res response object
+    * @param {object} next next
+    * @memberof Exists
+    * @returns {object} data
+    */
+  static async isUsersRequestExist(req, res, next) {
```
Comment (ivymwende) [inline src/middlewares/Exists.js]: @hezronkimutai Please explain how this is different from the database query you made in your controller and it's importance?

**Review by NiyoEric — APPROVED** [review]
You have done a good job @hezronkimutai.

**Comment by NiyoEric** [comment]
Update your README @hezronkimutai  to help people to test the feature.

**Review by ivymwende — CHANGES_REQUESTED** [review]
Great job! @hezzie Great Job! I have however left a few comments.

**Review by ivymwende — COMMENTED** [review]
@hezronkimutai Great Job!! Please clarify on the comment i left

File: src/tests/060-tripTest.js [hunk 4]
```diff
@@ -214,3 +214,50 @@ describe('Request One way trip test', () => {
...
+      .end((err, res) => {
+        res.should.have.status(401);
+        res.body.should.be.an('object');
+        done();
+      });
+  });
+  it('A user should be able to get his/her trip requests', (done) => {
+    chai.request(app).get('/api/v1/trips/')
+      .end((err, res) => {
+        res.should.have.status(401);
```
Comment (ivymwende) [inline src/tests/060-tripTest.js:259]: @hezronkimutai Why us the status code 401 if the user is able to get his/her trip requests?

File: src/middlewares/Exists.js (hunk 3 above)
Comment (hezronkimutai) [reply src/middlewares/Exists.js]: This is a middleware meant to check if the user has any  requests,   the querry at the controller is a join of two tables. get all requests made by the user.

**Review by ivymwende — APPROVED** [review]

File: src/routes/api/tripRoute.js [hunk 5]
```diff
@@ -239,4 +239,30 @@ tripRouter.post('/twoWay',
...
+ *          description: enter token
+ *          required: true
+ *          schema:
+ *            type: string
+ *      responses:
+ *        "200":
+ *          description: Trip requests fetched successfuly
+ *        "404":
+ *          description: Trip requests are not found
+ */
```
Comment (victkarangwa) [inline src/routes/api/tripRoute.js:269]: Would do you mind adding a response for users without tokens?
like 
```
"401": 
description: No token provided
```

**Review by victkarangwa — COMMENTED** [review]
@hezronkimutai  Nice Job
Just have a look at a comment I've left

**Review by victkarangwa — APPROVED** [review]
@hezronkimutai Thanks for incorporating the feedback.
I think you are good to go!

**Pull request merged**